virtio-devices/src/block.rs

```diff
@@ -690,9 +690,19 @@ impl Block {
                 (disk_nsectors, avail_features, 0, config, false)
             };
 
-        let serial = serial
-            .map(Vec::from)
-            .unwrap_or_else(|| build_serial(&disk_path));
+        let serial = if let Some(s) = serial {
+            let mut s = Vec::from(s);
+            if s.len() > VIRTIO_BLK_ID_BYTES as usize {
```

Comment:
May be we should not let user create the device, we are modifying the content provided by the user, Who know user might use this value to track something?